mm2src/coins/tendermint/iris/htlc.rs

```diff
@@ -0,0 +1,191 @@
+// IRIS HTLC implementation in Rust on top of Cosmos SDK(cosmrs) for AtomicDEX.
+//
+// This module includes HTLC creating & claiming representation structstures
+// and their trait implementations.
+
+use cosmrs::{tx::{Fee, Msg, MsgProto},
+             AccountId, Coin, ErrorReport};
+use std::convert::TryFrom;
+
+const CREATE_HTLC_TYPE_URL: &str = "/irismod.htlc.MsgCreateHTLC";
+const CLAIM_HTLC_TYPE_URL: &str = "/irismod.htlc.MsgClaimHTLC";
+
+#[allow(dead_code)]
+pub(crate) struct IrisHtlc {
+    /// Generated HTLC's ID.
+    pub(crate) id: String,
+
+    /// Transaction fee
+    pub(crate) fee: Fee,
+
+    /// Message payload to be sent
+    pub(crate) msg_payload: cosmrs::Any,
+}
+
+/// Proto representation of create HTLC message
+#[derive(prost::Message)]
+pub(crate) struct CreateHtlcProtoRep {
```

Comment:
Additionally to this comment, could you please move the protobuf structure definitions to a separate file? For example, `htlc_proto.rs`?

Reply:
> Is there a proto file that we can use/compile instead?

[Yes, here is](https://github.com/irisnet/irismod/blob/5016c1be6fdbcffc319943f33713f4a057622f0a/proto/htlc/htlc.proto). But for this, we have to create multiple proto files because they depend on each other. Which will generate a lot than we we actually need.

> Additionally to this comment, could you please move the protobuf structure definitions to a separate file? For example, htlc_proto.rs?

Sure, will do thanks.

Reply:
> Which will generate a lot than we we actually need.

Agreed, way too many additional imports there, thanks for the info! 🙂

Reply:
> could you please move the protobuf structure definitions to a separate file? 

Done